pkg/controller/networkpolicy/crd_utils.go

```diff
@@ -131,14 +133,46 @@ func toAntreaIPBlockForCRD(ipBlock *crdv1beta1.IPBlock) (*controlplane.IPBlock,
 	if err != nil {
 		return nil, err
 	}
+	var exceptNets []controlplane.IPNet
+	for _, exc := range ipBlock.Except {
+		// Convert the except IPBlock to networkpolicy.IPNet.
+		exceptNet, err := cidrStrToIPNet(exc)
+		if err != nil {
+			return nil, err
+		}
+		exceptNets = append(exceptNets, *exceptNet)
+	}
 	antreaIPBlock := &controlplane.IPBlock{
-		CIDR: *ipNet,
-		// secv1alpha.IPBlock does not have the Except slices.
-		Except: []controlplane.IPNet{},
+		CIDR:   *ipNet,
+		Except: exceptNets,
 	}
 	return antreaIPBlock, nil
 }
 
+// computeEffectiveIPNetForIPBlocks calculates the list of net.IPNet CIDRs after the
+// "except" CIDRs are subtracted from each corresponding ipBlock.
+func computeEffectiveIPNetForIPBlocks(ipBlocks []crdv1beta1.IPBlock) []*net.IPNet {
```

Comment:
this function should be unit tested independently

Reply:
Done. See the latest commit